modules/Bio/EnsEMBL/Compara/Utils/Cigars.pm

```diff
@@ -634,4 +706,241 @@ sub check_cigar_line {
       if ($seq_pos != $length);
 }
 
+
+=head2 compute_alignment_depth
+
+  Arg [1]    : Array-ref of Strings $cigar_lines
+  Arg [2]    : (optional) Array-ref of names (identifiers) giving the group of each cigar-line
+  Example    : compute_alignment_depth($cigar_lines, $genome_db_ids);
+  Description: Returns statistics about the alignment depth (number of aligned sequences) for each sequence. The
+               function returns for each sequence its number of positions (total and aligned), how many positions
+               have each depth level, and the sum of all the depths.
+               Sequences can be grouped, in which case the statistics are returned by group (not by sequence) and the
+               depth represents the number of aligned groups.
+  Returntype : Hash-ref {id => { n_total_pos => XXX, n_aligned_pos => YYY, depth_sum => ZZZ, depth_breakdown => { 0 => aaa, 1 => bbb, ...}}}
+  Exceptions : none
+
+=cut
+
+sub compute_alignment_depth {
+    my $cigar_lines = shift;
+    my $group_ids = shift;
+
+    my $n_cigars = scalar(@$cigar_lines);
+
+    # If no groups are required, consider each sequence individually
+    unless ($group_ids) {
+        $group_ids = [0..($n_cigars-1)];
+    }
+
+    my %depth_sum;
+    my %depth_breakdown;
+    my %n_total_pos;
+    my %n_aligned_pos;
+
+    my $cb = sub {
+        my ($pos, $codes, $length) = @_;
+
+        # Will contain the group_ids that are present on these columns
+        # and the number of sequences they contain
+        my %n_aligned_ids;
+        for (my $i = 0; $i < $n_cigars; $i++ ) {
+            if ( $codes->[$i] eq 'M' ) {
+                $n_aligned_ids{ $group_ids->[$i] } ++;
+            }
+        }
+        # "- 1" because the depth is the number of *other* sequences
+        my $this_depth = scalar(keys %n_aligned_ids) - 1;
```

Comment:
Should it be other sequences (as the comment suggests) or other groups (as the code suggests)?

Reply:
Other groups indeed